web/src/main/java/org/mskcc/cbio/oncokb/controller/DriveAnnotationParser.java

```diff
@@ -513,53 +518,38 @@ private static void parseCancer(Gene gene, Set<Alteration> alterations, JSONObje
 //                evidence.setAdditionalInfo(additionalInfo);
 //            }
 
             if (cancerObj.has("prevalence") && !cancerObj.getString("prevalence").trim().isEmpty()) {
-                System.out.println("###         Full prevalence: " + alterations.toString());
+                System.out.println(spaceStrByNestLevel(nestLevel + 2) + "Has description.");
                 String desc = cancerObj.getString("prevalence").trim();
                 evidence.setDescription(desc);
                 setDocuments(desc, evidence);
             }
             evidenceBo.save(evidence);
-        } else {
-            System.out.println("##      No Prevalance for " + alterations.toString());
         }
 
         // Prognostic implications
-        if ((cancerObj.has("progImp") && !cancerObj.getString("progImp").trim().isEmpty()) || (cancerObj.has("shortProgImp") && !cancerObj.getString("shortProgImp").trim().isEmpty())) {
-            System.out.println("##      Prognostic implications:" + alterations.toString());
-            Date lastEdit = cancerObj.has("progImp_review") ? getUpdateTime(cancerObj.get("progImp_review")) : null;
-            Evidence evidence = new Evidence();
-            evidence.setEvidenceType(EvidenceType.PROGNOSTIC_IMPLICATION);
-            evidence.setAlterations(alterations);
-            evidence.setGene(gene);
-            if (oncoTreeType.getMainType() != null) {
-                evidence.setCancerType(oncoTreeType.getMainType().getName());
-            }
-            evidence.setSubtype(oncoTreeType.getCode());
-            evidence.setUuid(cancerObj.has("progImp_uuid") ? cancerObj.getString("progImp_uuid") : "");
-            evidence.setLastEdit(lastEdit);
-//            if (cancerObj.has("shortProgImp") && !cancerObj.getString("shortProgImp").trim().isEmpty()) {
-//                System.out.println("###         Short prognostic implications: " + alterations.toString());
-//                String additionalInfo = cancerObj.getString("shortProgImp").trim();
-//                evidence.setAdditionalInfo(additionalInfo);
-//            }
-
-            if (cancerObj.has("progImp") && !cancerObj.getString("progImp").trim().isEmpty()) {
-                System.out.println("###         Full prognostic implications: " + alterations.toString());
-                String desc = cancerObj.getString("progImp").trim();
-                evidence.setDescription(desc);
-                setDocuments(desc, evidence);
-            }
-            evidenceBo.save(evidence);
-        } else {
-            System.out.println("##      No Proganostic implications " + alterations.toString());
-        }
+        parseImplication(gene, alterations, oncoTreeType,
+            cancerObj.has("prognostic") ? cancerObj.getJSONObject("prognostic") : null,
+            cancerObj.has("progImp_uuid") ? cancerObj.getString("progImp_uuid") : "",
+            cancerObj.has("progImp_review") ? getUpdateTime(cancerObj.get("progImp_review")) : null,
```

Comment:
This is not the right way to get the last edit time. We need to pass in a list of review objects, and pick the most recent one as the last edit time for Evidence. Take this one as example, we need to choose most recent update time among (cancerObj.prognostic.description_review, cancerObj.prognostic.level_review)